File: Random/3.Can Place Flowers.py
#    Author : Mohamed Yousef 
#    Date   : 2022-12-03
import sys,math,bisect,collections,itertools,heapq
from collections import defaultdict,deque
def solve(flower,n):
    x=len(flower)
    if x==1:
        if flower[0]==1:
            if n==0:return True
            return False
        return True
    if flower[0]==0 and flower[1]==0:
        flower[0]=1
        n-=1
    for i in range(1,x-1):
        if flower[i]==0 and flower[i+1]==0 and flower[i-1]==0:
            flower[i]=1
            n-=1
    if flower[-1]==0 and flower[-2]==0:
        flower[-1]=1
        n-=1
    if n<=0:return True
    return False

print(solve([1],0))

File: Random/3.Can Place Flowers.js
/*
    Author : Mohamed Yousef 
    Date   : 2022-12-03
*/
let solve=function(flowerbed,n){
    x=flowerbed.length
    if (x==1){
        if (flowerbed[0]==1) {
            if (n==0) return true
            return false
        }
        return true
    }
    if (flowerbed[0]==flowerbed[1]){
        flowerbed[0]=1
        n-=1
    }
    for(let i=1;i<x-1;i++){
        if(flowerbed[i]==0 && flowerbed[i+1]==0 && flowerbed[i-1]==0){
            n-=1
            flowerbed[i]=1
        }
    }
    if (flowerbed.at(-1)==flowerbed.at(-2)){
        flowerbed[0]=1
        n-=1
    }
    if (n<=0) return true
    return false
}
console.log(solve([1,0,0,0,1],1));